src/SessionManager/ProgramSettings.cc

```diff
@@ -34,6 +37,99 @@ void applyOptionalArgument(T& val, const string& argument_name, const cxxopts::P
+    }
+    ApplyCommandLineSettings(argc, argv);
+
+    const fs::path user_settings_path = fs::path(getenv("HOME")) / CARTA_USER_FOLDER_PREFIX / "backend.json";
+    const fs::path system_settings_path = "/etc/carta/backend.json";
+
+    json settings;
+    if (fs::exists(system_settings_path) && !no_system_config) {
+        spdlog::info("System settings found");
+        settings = JSONConfigSettings(system_settings_path.string());
+        system_settings_json_exists = true;
+    }
+
+    if (fs::exists(user_settings_path) && !no_user_config) {
+        if (system_settings_json_exists) {
+            spdlog::info("User settings found, applying on top of system settings");
+        } else {
+            spdlog::info("User settings found");
+        }
+        auto user_settings = JSONConfigSettings(user_settings_path.string());
+        user_settings_json_exists = true;
+        settings.merge_patch(user_settings); // user on top of system
+    }
+
+    if (system_settings_json_exists || user_settings_json_exists) {
+        settings.merge_patch(command_line_settings); // force command-line on top of user and sytem
+        SetSettingsFromJSON(settings);
+    }
+}
+
+json ProgramSettings::JSONConfigSettings(const std::string& json_file_path) {
+    std::ifstream ifs(json_file_path, std::ifstream::in);
+    json j;
+    try {
+        j = json::parse(ifs);
+    } catch (json::exception err) {
+        spdlog::warn("Config file {} has problems, please check", json_file_path);
+        spdlog::warn(err.what());
+    }
+    return j;
+}
+
+void ProgramSettings::SetSettingsFromJSON(const json& j) {
+    if (j.contains("verbosity")) {
+        verbosity = j.at("verbosity");
+    }
+    if (j.contains("no_log")) {
+        no_log = j.at("no_log");
+    }
+    if (j.contains("log_performance")) {
+        log_performance = j.at("log_performance");
+    }
+    if (j.contains("log_protocol_messages")) {
+        log_protocol_messages = j.at("log_protocol_messages");
+    }
+    if (j.contains("no_http")) {
+        no_http = j.at("no_http");
+    }
+    if (j.contains("no_browser")) {
+        no_browser = j.at("no_browser");
+    }
+    if (j.contains("host")) {
+        host = j.at("host");
+    }
+    if (j.contains("port")) {
+        port = j.at("port");
+    }
+    if (j.contains("grpc_port")) {
+        grpc_port = j.at("grpc_port");
+    }
+    if (j.contains("omp_threads")) {
+        omp_thread_count = j.at("omp_threads");
+    }
+    if (j.contains("top_level_folder")) {
+        top_level_folder = j.at("top_level_folder");
+    }
+    if (j.contains("frontend_folder")) {
+        frontend_folder = j.at("frontend_folder");
+    }
+    if (j.contains("exit_timeout")) {
+        wait_time = j.at("exit_timeout");
+    }
+    if (j.contains("initial_timeout")) {
+        init_wait_time = j.at("initial_timeout");
+    }
+    if (j.contains("idle_timeout")) {
+        idle_session_wait_time = j.at("idle_timeout");
+    }
```

Comment:
I agree with both of you. I implemented it both ways, and of course, using `value` is shorter, but it becomes harder to guess what's happening when you read it for the first time. I think it is more evident to know what's going on when you read:
```
    if (j.contains("verbosity")) {
        verbosity = j.at("verbosity");
    }
```
rather than:
```
verbosity = j.value("verbosity", verbosity);
```
without going to the reference/documentation.

I will use `value`.